ui/app/actions.js

```diff
@@ -857,3 +833,24 @@ function shapeShiftRequest (query, options, cb) {
     return shapShiftReq.send()
   }
 }
+
+// Call Background Then Update
+//
+// A function generator for a common pattern wherein:
+// We show loading indication.
+// We call a background method.
+// We hide loading indication.
+// If it errored, we show a warning.
+// If it didn't, we update the state.
+function callBackgroundThenUpdate (method, ...args) {
+  return (dispatch) => {
+    dispatch(actions.showLoadingIndication())
+    method.call(background, ...args, (err, newState) => {
+      dispatch(actions.hideLoadingIndication())
+      if (err) {
+        return dispatch(actions.displayWarning(err.message))
```

Comment:
There may be a problem here with using displayWarning instead of this.showWarning?